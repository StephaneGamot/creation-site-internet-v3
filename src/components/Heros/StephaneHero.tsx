"use client"

import Image from 'next/image'
import { motion } from 'framer-motion'
import Portrait from "./../../../public/images/stephane.webp"
import { useTranslations } from 'next-intl'

export default function StephaneHero() {
  const t = useTranslations("stephaneHero");
  const lines: string[] = t.raw("lines");

  return (
    <section aria-labelledby="stephane-hero-title" className="max-w-5xl mx-auto grid md:grid-cols-2 gap-12 items-center">
      <motion.div
        initial={{ opacity: 0, x: -30 }}
        animate={{ opacity: 1, x: 0 }}
        transition={{ duration: 0.6 }}
      >
        <Image
          src={Portrait}
          alt={t("alt")}
          className="rounded-2xl shadow-xl"
        />
      </motion.div>

      <motion.div
        initial={{ opacity: 0, x: 30 }}
        animate={{ opacity: 1, x: 0 }}
        transition={{ duration: 0.6 }}
        className="space-y-4"
      >
        <h1 id="stephane-hero-title" className="text-4xl md:text-5xl font-title font-semibold">
          {t("title")}
        </h1>
        {lines.map((line, index) => (
          <p
            key={index}
            className={`text-gray-700 ${index === lines.length - 1 ? 'text-sm' : 'text-lg'} leading-relaxed`}
          >
            {line}
          </p>
        ))}
      </motion.div>
    </section>
  );
}
