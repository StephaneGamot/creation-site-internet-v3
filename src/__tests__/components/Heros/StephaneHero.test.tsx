import React from "react";
import { render, screen } from "@testing-library/react";
import StephaneHero from "@/components/Heros/StephaneHero";
import { axe } from "jest-axe";
import renderer from "react-test-renderer";

// ✅ Mock next-intl
jest.mock("next-intl", () => ({
  useTranslations: () => {
    const t = (key: string) => {
      const values: Record<string, string> = {
        title: "Qui suis-je ?",
        alt: "Portrait de Stéphane Gamot",
      };
      return values[key] || "";
    };
    t.raw = (key: string) => {
      if (key === "lines") {
        return [
          "Développeur passionné, créatif et rigoureux.",
          "J’allie technique et esthétique pour des sites performants.",
          "Basé en Belgique, disponible pour vos projets web.",
        ];
      }
      return [];
    };
    return t;
  },
}));

// ✅ Mock framer-motion
jest.mock("framer-motion", () => ({
  motion: {
    div: (props: any) => <div {...props} />,
  },
}));

// ✅ Mock next/image
jest.mock("next/image", () => (props: any) => {
  return <img {...props} alt={props.alt || "mocked image"} />;
});

describe("StephaneHero", () => {
  it("affiche le titre, l’image et les lignes de texte", () => {
    render(<StephaneHero />);
    expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent("Qui suis-je ?");
    expect(screen.getByAltText("Portrait de Stéphane Gamot")).toBeInTheDocument();

    expect(screen.getByText("Développeur passionné, créatif et rigoureux.")).toBeInTheDocument();
    expect(screen.getByText("J’allie technique et esthétique pour des sites performants.")).toBeInTheDocument();
    expect(screen.getByText("Basé en Belgique, disponible pour vos projets web.")).toBeInTheDocument();
  });

  it("passe les tests d’accessibilité", async () => {
    const { container } = render(<StephaneHero />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it("correspond au snapshot", () => {
    const tree = renderer.create(<StephaneHero />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
