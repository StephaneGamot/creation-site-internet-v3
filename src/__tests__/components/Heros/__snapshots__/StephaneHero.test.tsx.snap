// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StephaneHero correspond au snapshot 1`] = `null`;
